core/domain/wipeout_service.py

```diff
@@ -109,11 +178,115 @@ def _delete_user_models(user_id):
     """
     for model_class in models.Registry.get_storage_model_classes(
             [models.NAMES.user]):
-        if (model_class.get_deletion_policy() !=
-                base_models.DELETION_POLICY.KEEP):
+        if (model_class.get_deletion_policy() not in
+                [base_models.DELETION_POLICY.KEEP,
+                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
             model_class.apply_deletion_policy(user_id)
 
 
+def _hard_delete_explorations_and_collections(pending_deletion):
```

Comment:
Should call the arg pending_deletion_request, here and elsewhere (above as well). Don't use "pending_deletion" as a substitute -- that sounds more like an adjective than a noun.